unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs

```diff
@@ -262,9 +280,22 @@ void UpdateStats()
             {
                 avgFullProcessLatencyText.text = (avgFullProcessLatency * 1000).ToString("f0") + " ms";
                 avgServerProcessingTimeText.text = (avgServerProcessingTime).ToString("f0") + " ms";
-                avgNetworkOnlyLatencyText.text = (avgNetworkOnlyLatency).ToString("f0") + " ms";
+
             }
         }
+
+       async void GetNetworkOnlyLatency()
+        {
+            HttpClient httpClient = new HttpClient();
+            HttpRequestMessage request =new HttpRequestMessage(HttpMethod.Head,new Uri(UriBasedOnConnectionMode()));
+            System.Diagnostics.Stopwatch timer = new System.Diagnostics.Stopwatch();
+            timer.Start();
+            HttpResponseMessage response = await httpClient.SendAsync(request);
+            timer.Stop();
```

Comment:
This says "This operation will not block": https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=netcore-3.1 

I don't know. Maybe it is different for Unity? I see SendAsync(), and I think "fire this off and execute the next line of code immediately." If this is the case, the elapsed time would be super short. Even to Germany, this would probably show 1 ms or less. 

Presumably, you have tested this and actually see different, reasonable results? If so, ignore this comment, and I need to learn what "Async" means in Unity.

Reply:
I should have mentioned that the HEAD /test/ endpoint only exists in the latest ComputerVisionServer code, and won't be available yet in the existing deployed app instances. HOWEVER, I think the call to a non-existent endpoint will still work and be an accurate measurement. You'll just get a "404 Not Found" instead of a "200 OK". Even though it's an error response, It's still a round trip time to the server and back.

Reply:
I am using Unity Web request in a **coroutine**, UnityWebRequest is basically a SendAsync at its low level.
yield return basically means stop reading the script and continue to the next frame.

it can be yield return WaitForSeconds or yield return for Another Coroutine , In this case I am waiting for the web request.

You have a point though I did it with Http.SendAsync in a regular function and it was blocking Unity main thread, Thats why I guess Coroutines are the best way to run asynchronous operations in Unity Single thread.

I will post a video on slack for the performance, Thanks for your thorough review.

Reply:
It's await Task essentially. Depends on thread context switch time (a few ms/quanta), and how busy it gets. Coroutine would be tied to frame rate between yields...measure and pick one that works better (in general).